realm/realm-library/src/syncIntegrationTest/java/io/realm/objectserver/SyncSessionTests.java

```diff
@@ -455,4 +462,121 @@ public void run() {
+                .modules(new PartialSyncModule())
+                .build();
+
+        // Using Admin user, populate the Realm.
+        Realm realm = Realm.getInstance(adminConfig);
+        realm.beginTransaction();
+        PartialSyncObjectA objectA = realm.createObject(PartialSyncObjectA.class);
+        objectA.setNumber(0);
+        objectA.setString("realm");
+        objectA = realm.createObject(PartialSyncObjectA.class);
+        objectA.setNumber(1);
+        objectA.setString("");
+        objectA = realm.createObject(PartialSyncObjectA.class);
+        objectA.setNumber(2);
+        objectA.setString("");
+        objectA = realm.createObject(PartialSyncObjectA.class);
+        objectA.setNumber(3);
+        objectA.setString("");
+        objectA = realm.createObject(PartialSyncObjectA.class);
+        objectA.setNumber(4);
+        objectA.setString("realm");
+        objectA = realm.createObject(PartialSyncObjectA.class);
+        objectA.setNumber(5);
+        objectA.setString("sync");
+        objectA = realm.createObject(PartialSyncObjectA.class);
+        objectA.setNumber(6);
+        objectA.setString("partial");
+        objectA = realm.createObject(PartialSyncObjectA.class);
+        objectA.setNumber(7);
+        objectA.setString("partial");
+        objectA = realm.createObject(PartialSyncObjectA.class);
+        objectA.setNumber(8);
+        objectA.setString("partial");
+        objectA = realm.createObject(PartialSyncObjectA.class);
+        objectA.setNumber(9);
+        objectA.setString("partial");
+
+        realm.createObject(PartialSyncObjectB.class).setNumber(0);
+        realm.createObject(PartialSyncObjectB.class).setNumber(1);
+        realm.createObject(PartialSyncObjectB.class).setNumber(2);
+        realm.createObject(PartialSyncObjectB.class).setNumber(3);
+        realm.createObject(PartialSyncObjectB.class).setNumber(4);
+        realm.createObject(PartialSyncObjectB.class).setNumber(5);
+        realm.createObject(PartialSyncObjectB.class).setNumber(6);
+        realm.createObject(PartialSyncObjectB.class).setNumber(7);
+        realm.createObject(PartialSyncObjectB.class).setNumber(8);
+        realm.createObject(PartialSyncObjectB.class).setNumber(9);
+        realm.commitTransaction();
+
+        SyncManager.getSession(adminConfig).uploadAllLocalChanges();
+        realm.close();
+
+        final CountDownLatch latch = new CountDownLatch(2);
+
+        HandlerThread handlerThread = new HandlerThread("background");
+        handlerThread.start();
+        Handler handler = new Handler(handlerThread.getLooper());
+        handler.post(new Runnable() {
+            @Override
+            public void run() {
+                final Realm partialSyncRealm = Realm.getInstance(partialSyncConfig);
+                assertTrue(partialSyncRealm.isEmpty());
+
+                partialSyncRealm.subscribeToObjects(PartialSyncObjectA.class, "number > 5", new Realm.PartialSyncCallback<PartialSyncObjectA>() {
```

Comment:
This is the first time I see string based query API in Realm-Java, I don't know if its usage is documented anywhere.

Reply:
@Zhuinden This is a preview of a upcoming feature, and we might not have documented completely yet.

Reply:
Not what I meant. I'm just surprised this isn't based on `RealmQuery<T>` which pretty much does the same thing except not as a String.

Reply:
It probably will be. This is just a really early preview of it in order to facilitate feedback. The API _will_ change.